src/sdk/autodetectcompilers.cpp

```diff
@@ -40,109 +45,240 @@ AutoDetectCompilers::AutoDetectCompilers(wxWindow* parent)
-    if (idx != -1)
+    int idxList = list->GetNextItem(-1, wxLIST_NEXT_ALL, wxLIST_STATE_SELECTED);
+    if (idxList != -1)
+    {
+        wxString wsSelection = list->GetItemText(idxList);
+        Compiler* compiler = CompilerFactory::GetCompilerByName(wsSelection);
+        int idxComiler = CompilerFactory::GetCompilerIndex(compiler);
+
+        CompilerFactory::SetDefaultCompiler(idxComiler);
+
+        wxColour  colorDefault =  XRCCTRL(*this, "ID_STATICTEXT1", wxStaticText)->GetBackgroundColour();
+        wxStaticText* controlLblDefCompiler = XRCCTRL(*this, "lblDefCompiler", wxStaticText);
+        controlLblDefCompiler->SetLabel(CompilerFactory::GetDefaultCompiler()->GetName());
+        controlLblDefCompiler->SetBackgroundColour(colorDefault);
+    }
+}
+
+void AutoDetectCompilers::OnUdateCompilerListUI(cb_unused wxCommandEvent& event)
+{
+    UpdateCompilerDisplayList();
+}
+
+void AutoDetectCompilers::UpdateCompilerDisplayList()
+{
+    bool bShowCompilerPath= XRCCTRL(*this,"cbShowCompilerPath", wxCheckBox)->GetValue();
+    bool bShowAllCompilerOptions = (XRCCTRL(*this, "rbCompilerShowOptions", wxRadioBox)->GetSelection() == 1);
+
+    wxListCtrl* list = XRCCTRL(*this, "lcCompilers", wxListCtrl);
+    if (list)
     {
-        CompilerFactory::SetDefaultCompiler(idx);
-        XRCCTRL(*this, "lblDefCompiler", wxStaticText)->SetLabel(CompilerFactory::GetDefaultCompiler()->GetName());
+        list->ClearAll();
+        list->InsertColumn(ccnNameColumn, _("Compiler"), wxLIST_FORMAT_LEFT, 380);
+        list->InsertColumn(ccnStatusColumn, _("Status"), wxLIST_FORMAT_LEFT, 100);
+
+        if (bShowCompilerPath)
+            list->InsertColumn(ccnDetectedPathColumn, _("Compiler Path"), wxLIST_FORMAT_LEFT, 200);
+
+        for (auto itCL = std::begin(vCompilerList); itCL != std::end(vCompilerList); ++itCL)
+        {
+            if (bShowAllCompilerOptions && !itCL->bDetected)
+                continue;
+
+            int idx = list->GetItemCount();
+            list->InsertItem(idx, itCL->wxsCompilerName);
+            list->SetItem(idx, ccnStatusColumn, itCL->wxsStatus);
+            if (bShowCompilerPath)
+                list->SetItem(idx, ccnDetectedPathColumn, itCL->wxsCompilerPath);
+
+            switch(itCL->eHighlight)
+            {
+            case CompilerHighlightColor::eHighlightGrey:
+                    list->SetItemTextColour(idx, *wxLIGHT_GREY);
+                    break;
+                case CompilerHighlightColor::eHighlightGreen:
+                    list->SetItemBackgroundColour(idx, *wxGREEN);
+                    break;
+                case CompilerHighlightColor::eHighlightRed:
+                    list->SetItemBackgroundColour(idx, *wxRED);
+                    break;
+                case CompilerHighlightColor::eHighlightYellow:
+                    list->SetItemBackgroundColour(idx, *wxYELLOW);
+                    break;
+                case CompilerHighlightColor::eHighlightNone:

```

Comment:
Typical offence - set the BG color without setting the FG colour or vice-versa. Hardcoding colours in this dialog is wrong idea. It doesn't probably work well with dark-themes or red-themes or green-themes. There are strange people around the world, please don't make them suffer even more.

I think it is best to remove any colouring from this dialog. Colours could be used in other places, but the cbColourManager should be available. Here it is not.

Reply:
100% agree with this as a typical offence when I searched for ways to fix it.

How in this area how do I make it easier for the user to see which compiler has a good or bad path or the autodetect code has detected the path has changed?

Reply:
> How in this area how do I make it easier for the user to see ...

Something that is reliable in a themeable world.

>...  or the autodetect code has detected the path has changed?

Why would you want to show this?
When does it happen?

Reply:
> > How in this area how do I make it easier for the user to see ...
>
> Something that is reliable in a themeable world.

So what is reliable in the themeable world with the current C::B code? This dialog only pops up on first install or if something changes with the default compiler on subsequent startups. For the vast majority of users it will only pop up once on the initial install. I am the exception at the moment due to the testing and swapping in/out different compilers for testing purposes in order to text the various combinations and the pop up itself.


> > ...  or the autodetect code has detected the path has changed?
>
> Why would you want to show this?

I want to show this to confirm to the end user that the compiler directory change is okay and acceptable instead of C::B  automatically changing the directory on startup without letting the end user know that it occurred.

> When does it happen?

This occurs when the end users moves or renames a compiler directory on their C: or when they switch in/out different compilers on their C: (there are so many different MingW versions out there that I expect this will occur more that the move/rename).

Reply:
Any updates?

Reply:
> I want to show this to confirm to the end user that the compiler directory change is okay and acceptable instead of C::B automatically changing the directory on startup without letting the end user know that it occurred.

This is wrong. C::B must not detect changes to the compiler after initial setup. If the compiler setup on the machine changes, I think people should be expected to update their settings.

> This occurs when the end users moves or renames a compiler directory on their C: or when they switch in/out different compilers on their C: (there are so many different MingW versions out there that I expect this will occur more that the move/rename).

I don't think this case should be handled at all. The auto-detect dialog should be shown only to people which don't have default.conf. Nobody expects or wants this dialog to be shown. This adds just useless and annoying complexity to the code.